code/go/0chain.net/chaincore/transaction/worker.go

```diff
@@ -91,8 +91,45 @@ func RemoveFromPool(ctx context.Context, txns []datastore.Entity) {
 	txn := transactionEntityMetadata.Instance().(*Transaction)
 	collectionName := txn.GetCollectionName()
 
+	logging.Logger.Debug("cleaning invalid transactions")
+	mappedTxns := make(map[string][]*Transaction)
+	for _, e := range txns {
+		blockTx, ok := e.(*Transaction)
+		if !ok {
+			logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", e))
+			continue
+		}
+		mappedTxns[blockTx.GetKey()] = append(mappedTxns[blockTx.GetKey()], blockTx)
+	}
+
+	var past []datastore.Entity
+	err := transactionEntityMetadata.GetStore().IterateCollection(cctx, transactionEntityMetadata, collectionName,
+		func(ctx context.Context, qe datastore.CollectionEntity) bool {
+			current, ok := qe.(*Transaction)
+			if !ok {
+				logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", qe))
+				return true
+			}
+
+			list := mappedTxns[txn.GetKey()]
+			for _, fromList := range list {
+				if fromList.GetKey() == current.GetKey() {
+					if current.Nonce <= fromList.Nonce {
```

Comment:
The txn.GetKey() returns the transaction hash, so the nonce would always be the same as they are the same transaction.